Sources/NIOPosix/Bootstrap.swift

```diff
@@ -482,6 +483,454 @@ public final class ServerBootstrap {
+        isChildChannelOutboundHalfClosureEnabled: Bool = false
+    ) async throws -> NIOAsyncChannel<NIOAsyncChannel<ChildChannelInbound, ChildChannelOutbound>, Never> {
+        func makeChannel(_ eventLoop: SelectableEventLoop, _ childEventLoopGroup: EventLoopGroup, _ enableMPTCP: Bool) throws -> ServerSocketChannel {
+            if enableMPTCP {
+                throw ChannelError.operationUnsupported
+            }
+            return try ServerSocketChannel(socket: socket, eventLoop: eventLoop, group: childEventLoopGroup)
+        }
+        return try await self.bindAsyncChannel0(
+            makeServerChannel: makeChannel,
+            serverBackpressureStrategy: serverBackpressureStrategy,
+            childBackpressureStrategy: childBackpressureStrategy,
+            isChildChannelOutboundHalfClosureEnabled: isChildChannelOutboundHalfClosureEnabled
+        ) { (eventLoop, serverChannel) in
+            let promise = eventLoop.makePromise(of: Void.self)
+            serverChannel.registerAlreadyConfigured0(promise: promise)
+            return promise.futureResult
+        }
+    }
+
+    @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+    private func bindAsyncChannel0<ChildChannelInbound: Sendable, ChildChannelOutbound: Sendable>(
+        serverBackpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark?,
+        childBackpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark?,
+        isChildChannelOutboundHalfClosureEnabled: Bool,
+        _ makeSocketAddress: () throws -> SocketAddress
+    ) async throws -> NIOAsyncChannel<NIOAsyncChannel<ChildChannelInbound, ChildChannelOutbound>, Never> {
+        let address = try makeSocketAddress()
+
+        func makeChannel(_ eventLoop: SelectableEventLoop, _ childEventLoopGroup: EventLoopGroup, _ enableMPTCP: Bool) throws -> ServerSocketChannel {
+            return try ServerSocketChannel(eventLoop: eventLoop,
+                                           group: childEventLoopGroup,
+                                           protocolFamily: address.protocol,
+                                           enableMPTCP: enableMPTCP)
+        }
+
+        return try await self.bindAsyncChannel0(
+            makeServerChannel: makeChannel,
+            serverBackpressureStrategy: serverBackpressureStrategy,
+            childBackpressureStrategy: childBackpressureStrategy,
+            isChildChannelOutboundHalfClosureEnabled: isChildChannelOutboundHalfClosureEnabled
+        ) { (eventLoop, serverChannel) in
+            serverChannel.registerAndDoSynchronously { serverChannel in
+                serverChannel.bind(to: address)
+            }
+        }
+    }
+
+    private typealias MakeServerChannel = (_ eventLoop: SelectableEventLoop, _ childGroup: EventLoopGroup, _ enableMPTCP: Bool) throws -> ServerSocketChannel
+    private typealias Register = (EventLoop, ServerSocketChannel) -> EventLoopFuture<Void>
```

Comment:
Do either of these need `@Sendable`?

Reply:
Yes `Register` needs to `Sendable` since it is run on the server socket EL. However, `MakeServerChannel` is run inline and not escaping.